From: "Barak A. Pearlmutter" <barak+git@cs.nuim.ie>
Date: Sun, 19 Oct 2014 16:29:06 +0100
Subject: define MAXHOSTNAMELEN

On GNU HURD, and possibly other platforms, MAXHOSTNAMELEN is not
defined as there is no fixed limit.  We crudely finesse this issue
with a default definition of 4096.  A proper solution would instead
involve dynamic allocation.
---
 src/ComTerp/comhandler.h       | 5 +++++
 src/Dispatch/rpcregistry.cc    | 6 ++++++
 src/OverlayUnidraw/aceimport.h | 5 +++++
 3 files changed, 16 insertions(+)

diff --git a/src/ComTerp/comhandler.h b/src/ComTerp/comhandler.h
index 62297d4..6300d46 100644
--- a/src/ComTerp/comhandler.h
+++ b/src/ComTerp/comhandler.h
@@ -42,6 +42,11 @@
 #include <ace/SOCK_Acceptor.h>
 #include <ace/Test_and_Set.h>
 
+// GNU HURD has no fixed limit
+#ifndef MAXHOSTNAMELEN
+#define MAXHOSTNAMELEN 4096
+#endif
+
 class ComTerpServ;
 
 //: An ACE_Test_and_Set Singleton.
diff --git a/src/Dispatch/rpcregistry.cc b/src/Dispatch/rpcregistry.cc
index b53d5ef..8c3d2bd 100644
--- a/src/Dispatch/rpcregistry.cc
+++ b/src/Dispatch/rpcregistry.cc
@@ -40,6 +40,12 @@
 #if defined(__linux__) || defined(__FreeBSD__)
 #include <stdio.h>
 #endif
+
+// GNU HURD has no fixed limit
+#ifndef MAXHOSTNAMELEN
+#define MAXHOSTNAMELEN 4096
+#endif
+
 // Print a short error message describing the last error encountered
 // during a call to a system function.
 
diff --git a/src/OverlayUnidraw/aceimport.h b/src/OverlayUnidraw/aceimport.h
index 8d827d7..ff526be 100644
--- a/src/OverlayUnidraw/aceimport.h
+++ b/src/OverlayUnidraw/aceimport.h
@@ -39,6 +39,11 @@ class OvImportCmd;
 #include <fstream.h>
 #include <iosfwd>
 
+// GNU HURD has no fixed limit
+#ifndef MAXHOSTNAMELEN
+#define MAXHOSTNAMELEN 4096
+#endif
+
 //: handler for import by socket into OverlayUnidraw.
 class UnidrawImportHandler : public ACE_Svc_Handler<ACE_SOCK_STREAM, ACE_NULL_SYNCH>
 {
